app/controllers/pages_controller.rb

```diff
@@ -4,6 +4,7 @@ class PagesController < ApplicationController
 
   def show
     @page = Page.find_by!(slug: params[:slug])
+    not_found unless FeatureFlag.enabled?(@page.feature_flag, current_user)
```

Comment:
This is just an idea, but what if we remove the feature flag name from the form and replace it with a check box that says "Protected by feature flag" or similar? We can then default the flag name to something like `page_#{slug}` which means we could perform this change without having to load a page record from the DB. Potentially we wouldn't even need to add a column to the `pages` table, just a method that returns the name from the slug. 

This would be easier for site admins (only one action instead of two), would save us an extra DB access and they only minor drawback would be that the flag name isn't fully configurable.

Reply:
I **love** this idea @citizen428! The only downsides are a) can't customize the flag name like you mentioned, and b) it wouldn't work if you had multiple pages you wanted guarded by a single feature flag.

Since I don't have those requirements for Codeland (which is driving this feature), I'm going to go for it!

Reply:
Sorry, I meant additional reads in the case the page is disabled. If we could check the flag independent of the page object, this read `@page = Page.find_by!(slug: params[:slug]` becomes unncessary in case the site is blocked by the flag. 

Anyway, you're probably right, like I said, it was just an idea 😃